### PR TITLE
Remove unnecessary newline in ConciseView

### LINKED ISSUE #11377: Unwanted empty lines in output of errors with $ErrorView=ConciseView
Unwanted empty lines in output of errors with $ErrorView=ConciseView

# Steps to reproduce

```powershell
# make sure it's the new default ConciseView
$ErrorView = 'ConciseView'

try {
    throw 'Oops!'
}
catch {
    $_
}
```

# Expected behavior

The error output has no unexpected and unwanted empty lines:

```none
Exception: C:\tmp\_191217_122216\2.ps1
Line |
   5 |  throw 'Oops!'
     |  ^ Oops!
```

# Actual behavior

The error output has an unexpected and unwanted empty line:

```none
Exception: C:\tmp\_191217_122216\2.ps1
Line |
   5 |  throw 'Oops!'

     |  ^ Oops!
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-rc.1
PSEdition                      Core
GitCommitId                    7.0.0-rc.1
OS                             Microsoft Windows 6.3.9600
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

While constructing the error message, there was an unnecessary newline between the script line and the error message.  Fix is to remove this newline.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/11377

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 1]
```diff
@@ -1103,7 +1103,7 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                                 $line = $line.Insert($offsetInLine + $offsetLength, $resetColor).Insert($offsetInLine, $accentColor)
```
Comment (PoshChan) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Invoke-WebRequest tests.Denial of service.Charset Parsing
```powershell
Expected the actual value to be greater than 10, but got 0.268694117874498.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1949
1949:             $pathologicalRatio | Should -BeGreaterThan 10
```

**Comment by SteveL-MSFT** [comment]
@PoshChan please retry linux

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-Linux`

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 2]
```diff
@@ -1103,7 +1103,7 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
...
                                             }
 
-                                            $posmsg += ""${accentColor}${lineWhitespace}${ScriptLineNumber} ${verticalBar} ${resetcolor}${line}`n""
+                                            $posmsg += ""${accentColor}${lineWhitespace}${ScriptLineNumber} ${verticalBar} ${resetcolor}${line}""
```
Comment (iSazonov) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: I see `$line = $myinv.Line` always ends with new line but I am not sure about ` $line = $_.TargetObject.LineText.Trim()` - can you confirm? Or maybe explicitly check and add newline if absent?

Comment (SteveL-MSFT) [reply src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: That's a good catch.  I'll add a check to add newline if absent.

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 3]
```diff
@@ -1097,6 +1097,9 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                                 $offsetInLine = $myinv.OffsetInLine - 1
                                             }
 
+                                            if (-not $line.EndsWith(""`n"")) {
+                                                $line += ""`n""
```
Comment (nightroman) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: Should not it be environment dependent new line? E.g. `System.Environment.NewLine`?
Output is redirected to files sometimes and then opened by Notepad which still does not like `\n`.

Comment (SteveL-MSFT) [reply src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: Good point.  Will update.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**